src/open_company_web/actions.cljs

```diff
@@ -0,0 +1,91 @@
+(ns open-company-web.actions
+  (:require [medley.core :as med]
+            [open-company-web.dispatcher :refer [action dispatch!]]
+            [open-company-web.lib.utils :as utils]
+            [open-company-web.router :as router]
+            [open-company-web.caches :as cache]
+            [open-company-web.api :as api]))
+
+(defn- delay-remove-loading []
+  (.setTimeout js/window #(dispatch! [:clear-loading]) 100))
+
+;; ---- Generic Actions Dispatch
+;; This is a small generic abstraction to handle "actions".
+;; An `action` is a transformation on the app state.
+;; The return value of an action will be used as the new app-state.
+
+;; The extended multimethod `action` is defined in the dispatcher
+;; namespace to avoid cyclical dependencies between namespaces
+
+(defmethod action :default [db payload]
+  (js/console.warn "No handler defined for" (str (first payload)))
+  (js/console.log "Full event: " (pr-str payload))
```

Comment:
we have a module that wraps the js console, is shodan: https://github.com/noprompt/shodan
no need to change this for now but good to keep it in mind for future work